pyproject.toml

```diff
@@ -5,7 +5,7 @@ description = ""
 authors = []
 
 [tool.poetry.dependencies]
-python = "^3.11"
+python = "~3.12"
```

Comment:
Do we want to bump or pin Python version in the public repository? If someone is using `youtrack-sdk`, they will have to upgrade to Python 3.12 even if they don't want/can't. I would prefer to use the lowest possible Python version. In this case, I would leave this repository on `^3.11`.

Reply:
This is a valid point, but I'm not sure if we want to make an exception here and have it different from the rest of our internal stuff. And I remember there were some resolution problems with `^` anyways... In this case maybe then something like `>3.11.0` ? Not sure which features we are actually using.

Reply:
> This is a valid point, but I'm not sure if we want to make an exception here and have it different from the rest of our internal stuff

We already made that decision when we made this repository public. We are responsible for what we have done.

> And I remember there were some resolution problems with ^ anyways...

I guess it was a problem with Renovate. But we've removed Python from the Renovate config. I don't know if we plan to use Renovate here.

> In this case maybe then something like >3.11.0 ? Not sure which features we are actually using.

We can leave it as is. `^3.11` covers `>=3.11,<4.0`. I would keep it like this before we'll know more about Python 4. We can also change the left side of the constraint when the lower Python version becomes obsolete.

Reply:
> > This is a valid point, but I'm not sure if we want to make an exception here and have it different from the rest of our internal stuff
> 
> We already made that decision when we made this repository public. We are responsible for what we have done.

We made the decision to make the repository public and provide our work as is to the community in the hope that it will be useful, but not to be responsible for user problems and provide any service beyond our best efforts :-) Caveat emptor = buyer beware...

> > And I remember there were some resolution problems with ^ anyways...
> 
> I guess it was a problem with Renovate. But we've removed Python from the Renovate config. I don't know if we plan to use Renovate here.

We definitively plan to use Renovate here. But I thought it was a problem with Poetry, not Renovate.

> > In this case maybe then something like >3.11.0 ? Not sure which features we are actually using.
> 
> We can leave it as is. `^3.11` covers `>=3.11,<4.0`. I would keep it like this before we'll know more about Python 4. We can also change the left side of the constraint when the lower Python version becomes obsolete.

We can use `>=3.11,<4.0` if Poetry only has problem with `^3.11`, but doesn't have a problem with `>=3.11,<4.0`.

Reply:
> provide any service beyond our best efforts

It's not a big deal not to upgrade the Python version if there is no real reason to do so.

> We can use >=3.11,<4.0 if Poetry only has problem with ^3.11, but doesn't have a problem with >=3.11,<4.0.

@marns93 could you please remind us what the problem was with `^3.11` specification?

Reply:
@catcombo sure, see detailed comment https://github.com/moneymeets/moneymeets-pulumi/pull/448#issuecomment-1689750530

Reply:
> @catcombo sure, see detailed comment https://github.com/moneymeets/moneymeets-pulumi/pull/448#issuecomment-1689750530

Thank you! @zyv In one of our other API (see PR #551) we added Python to ignore list in the Renovate config. So we can do the same here and keep `^3.11`. Is it correct, @marns93 ?

Reply:
@marns93 Please explain it yo Yury

Reply:
@catcombo I don't understand what should I explain to Yury. As he mentioned, this problem has nothing to do with Renovate. If you don't specify it with `~3.12` `dataclass-json` will be resolved to a different version even on your local machine. The ignore list for Renovate is used to skip Python updates itself, because they are performed manually.

I see your point with this repository, but everybody can pin `youtrack-sdk` to older versions if they cannot update to Python `3.12`. I think this is also true for code changes which are available in newer Python versions.

So in general I would prefer to keep it consistent with all our other repositories.

If we need more discussions, let's talk after the daily. :)

Reply:
@marns93 Sorry, I misunderstood your previous explanation. Lets talk after the daily.

> I see your point with this repository, but everybody can pin youtrack-sdk to older versions if they cannot update to Python 3.12. I think this is also true for code changes which are available in newer Python versions.

And don't receive a `youtrack-sdk` updates until they upgrade to 3.12. There is not reason to upgrade this repository to Python 3.12. We don't plan to use any of 3.12 new features. This is a bad idea from my point of view.

Reply:
So in this meeting after the daily we decided to keep `~3.12`, because the only other way to support Python 3.11 and 3.12 without dependency resolution problems that come with `^3.11` is to specify the range always limiting to the latest unreleased version of Python like `>=3.11.0,<3.13.0` (instead of `^3.11` == `>=3.11.0,<4.0`), but this means:

1. Extra work for us and makes this repository different from our internal repositories
2. We introduce the same problems in the dependency resolution chain as other packages who do such smart things

Since this is a project we use internally and we make it available publicly with support on the best-effort basis in the hope that it will be useful to others, unless it's proven that we have more than 1 external user and there is strong demand to support previous versions of Python, we will keep it to the latest.

Use previous commit hashes if you need older Python.